src/main/java/gg/solarmc/clans/helper/PluginHelper.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    // Clan Invites to Player Methods
+
+    public boolean hasInvited(Clan clan, Player player) {
+        return invites.computeIfAbsent(clan.getClanId(), id -> getPlayerInviteCache())
+                .getIfPresent(player) != null;
+    }
+
+    public void addInvite(Clan clan, Player player) {
+        Cache<Player, Clan> playersInvited = invites.computeIfAbsent(clan.getClanId(), id -> getPlayerInviteCache());
+        playersInvited.put(player, clan);
+        invites.put(clan.getClanId(), playersInvited);
+    }
+
+    public void removeInvite(Clan clan, Player player) {
+        Cache<Player, Clan> playersInvited = invites.computeIfAbsent(clan.getClanId(), id -> getPlayerInviteCache());
+        playersInvited.invalidate(player);
+        invites.put(clan.getClanId(), playersInvited);
+    }
+
+    // Clan Ally Invites Methods
+
+    public boolean isClanPresentInAlly(Clan clan) {
+        return allyInvites.getIfPresent(clan) != null;
+    }
+
+    public boolean hasAllyInvited(Clan clan, Clan allyClan) {
+        return allyInvites.getIfPresent(clan).equals(allyClan);
+    }
+
+    public Clan getAllyInvite(Clan clan) {
+        return allyInvites.getIfPresent(clan);
+    }
+
+    public void addAllyInvite(Clan clan, Clan allyClan) {
+        allyInvites.put(clan, allyClan);
+    }
+
+    public void removeAllyInvite(Clan clan) {
+        allyInvites.invalidate(clan);
+    }
+
+    // Cache build Methods
+
+    private Cache<Player, Clan> getPlayerInviteCache() {
+        return getCache().evictionListener((solarPlayer, solarClan, cause) -> {
+            String name = ((Clan) solarClan).currentClanName();
+            // TODO: sendMsg -> leader -> Clan invitation to solarPlayer has expired
+            ((Player) solarPlayer).sendMessage(ChatColor.YELLOW + "Clan invitation from " + name + " has expired");
+        }).build();
+    }
+
+    private Caffeine<Object, Object> getCache() {
+        return Caffeine.newBuilder().expireAfterAccess(Duration.ofSeconds(60));
+    }
+
+    // Msg methods o:
+
+    public void sendNotInClanMsg(Player player) {
+        String[] msg = {
+                ChatColor.GOLD + "You are not in a Clan!",
+                "Join a clan using /clan join [ClanName] if you are invited :)",
+                "Create a clan using /clan create [ClanName]"
+        };
+
+        player.sendMessage(msg);
+    }
+
+    public void sendPlayerClanMsg(Server server, Player player, Clan clan, String prefix, String msg) {
+        sendClanMsg(server, clan,
+                Component.text(prefix + "> ", prefix.equals("clan") ? NamedTextColor.GOLD : NamedTextColor.LIGHT_PURPLE)
+                        .append(player.displayName())
+                        .append(Component.text(" " + msg)));
+    }
+
+    public void sendClanMsg(Server server, Clan clan, Component msg) {
+        DataCenter dataCenter = server.getDataCenter();
+
+        dataCenter.runTransact(transaction ->
```

Comment:
Exceptions are swallowed by the future; you need an `exceptionally` block at the end